test/e2e/storage/volume_io.go

```diff
@@ -0,0 +1,828 @@
+// Note: `fsizes` values are expected to each be at least 1kb and a multiple of 1024 bytes.
+// Note: nil can be passed for the podSecContext parm, in which case it is ignored.
+func testVolumeIO(f *framework.Framework, cs clientset.Interface, config framework.VolumeTestConfig, volsrc v1.VolumeSource, podSecContext *v1.PodSecurityContext, file string, fsizes []int64) (err error) {
+	writeBlk := strings.Repeat("abcdefghijklmnopqrstuvwxyz123456", 32) // 1k file for `dd`
+	volName := fmt.Sprintf("%s-%s", config.Prefix, "io-volume")
+	dir := path.Join("/opt", config.Prefix, config.Namespace)
+	dd_input := path.Join(dir, "dd_if")
+	// setup init environment and keep server-pod alive until it's deleted
+	initCmd := fmt.Sprintf("mkdir -p %s; echo %s >%s; sleep 3600", dir, writeBlk, dd_input)
+
+	clientPod := &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: config.Prefix + "-io-client",
+			Labels: map[string]string{
+				"role": config.Prefix + "-io-client",
+			},
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:  config.Prefix + "-io-client",
+					Image: framework.BusyBoxImage,
+					Command: []string{
+						"/bin/sh",
+						"-c",
+						initCmd,
+					},
+					VolumeMounts: []v1.VolumeMount{
+						{
+							Name:      volName,
+							MountPath: dir,
+						},
+					},
+				},
+			},
+			SecurityContext: podSecContext,
+			Volumes: []v1.Volume{
+				{
+					Name:         volName,
+					VolumeSource: volsrc,
+				},
+			},
+		},
+	}
+
+	By(fmt.Sprint("starting ", config.Prefix, " io-client"))
+	podsNamespacer := cs.CoreV1().Pods(config.Namespace)
+	clientPod, err = podsNamespacer.Create(clientPod)
+	if err != nil {
+		return fmt.Errorf("failed to create client pod %q: %v", clientPod.Name, err)
+	}
+	defer func() {
+		By(fmt.Sprintf("deleting client pod %q...", clientPod.Name))
+		delErr := framework.DeletePodWithWait(f, cs, clientPod)
+		if delErr != nil {
+			framework.Logf("client pod failed to delete: %v", delErr)
+			if err == nil { // delete err is returned if err is not set
+				err = delErr
+			}
+		}
+	}()
+
+	err = framework.WaitForPodRunningInNamespace(cs, clientPod)
+	if err != nil {
+		return fmt.Errorf("client pod %q not running: %v", clientPod.Name, err)
+	}
+
+	// create files of the passed-in file sizes
+	for _, fsize := range fsizes {
+		// file sizes must be a multiple of the length of the `writeBlk` value (1024 currently)
+		if fsize < int64(len(writeBlk)) {
+			fsize = int64(len(writeBlk))
+		} else if fsize/int64(len(writeBlk))*int64(len(writeBlk)) != fsize {
+			fsize = fsize/int64(len(writeBlk)) + int64(len(writeBlk))
+		}
+		fpath := path.Join(dir, fmt.Sprintf("%s-%d", file, fsize))
+
+		By(fmt.Sprintf("writing %d bytes to test file %s", fsize, fpath))
+		// dd's bs*seek = size (note this creates the file 1 byte more than the target size)
+		seek := fsize/int64(len(writeBlk)) - 1
```

Comment:
I think using seek in this way is incorrect and not going to generate IO.  The seek is just going to move the file pointer to 1kb before the end and is not going to write any data for the portion that it skipped.  So regardless of file size, this is only  writing 1kb of data.